### ISSUE TITLE
Unable to fill_null with a date or datetime value

### ISSUE THREAD

**Issue by olivier-lacroix** [pola-rs/polars#1755]
#### Are you using Python or Rust?

Python

#### What version of polars are you using?

0.10.19

#### What operating system are you using polars on?

Linux

#### Describe your bug.

`fill_null `fails on a column of type Date or Datetime

#### What are the steps to reproduce the behavior?

```python
import polars as pl
from datetime import datetime

s = pl.Series("A",[datetime.strptime("2021-01-01","%Y-%m-%d"),None])

s.fill_null(datetime.strptime("2021-01-01","%Y-%m-%d"))
```
#### What is the actual behavior?

```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_349/1187959143.py in <module>
----> 1 s.fill_null(start)

/opt/conda/lib/python3.9/site-packages/polars/eager/series.py in fill_null(self, strategy)
   2031         """
   2032         if not isinstance(strategy, str):
-> 2033             return self.to_frame().select(pl.col(self.name).fill_null(strategy))[
   2034                 self.name
   2035             ]

/opt/conda/lib/python3.9/site-packages/polars/lazy/expr.py in fill_null(self, fill_value)
    817 
    818         fill_value = expr_to_lit_or_expr(fill_value, str_to_lit=True)
--> 819         return wrap_expr(self._pyexpr.fill_null(fill_value._pyexpr))
    820 
    821     def fill_nan(self, fill_value: Union[str, int, float, "Expr"]) -> "Expr":

AttributeError: 'datetime.datetime' object has no attribute '_pyexpr'
```

#### What is the expected behavior?

```python
shape: (2,)
Series: 'A' [datetime]
[
	2021-01-01 00:00:00
	2021-01-01 00:00:00
]
```

**Comment by ritchie46** [comment]
You pass a python object. Try convert it to an expression with `pl.lit`

**Comment by olivier-lacroix** [comment]
@ritchie46 , I had tried that, it throws a different error:

```python
import polars as pl
from datetime import datetime

s = pl.Series("A",[datetime.strptime("2021-01-01","%Y-%m-%d"),None])

s.fill_null(pl.lit(datetime.strptime("2021-01-01","%Y-%m-%d")))
```

throws

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/tmp/ipykernel_799/2787361502.py in <module>
      4 s = pl.Series("A",[datetime.strptime("2021-01-01","%Y-%m-%d"),None])
      5 
----> 6 s.fill_null(pl.lit(datetime.strptime("2021-01-01","%Y-%m-%d")))

/opt/conda/lib/python3.9/site-packages/polars/eager/series.py in fill_null(self, strategy)
   2031         """
   2032         if not isinstance(strategy, str):
-> 2033             return self.to_frame().select(pl.col(self.name).fill_null(strategy))[
   2034                 self.name
   2035             ]

/opt/conda/lib/python3.9/site-packages/polars/lazy/expr.py in fill_null(self, fill_value)
    813                    * "zero"
    814         """
--> 815         if fill_value in ["backward", "forward", "min", "max", "mean", "zero", "one"]:
    816             return wrap_expr(self._pyexpr.fill_null_with_strategy(fill_value))
    817 

/opt/conda/lib/python3.9/site-packages/polars/lazy/expr.py in __bool__(self)
     79 
     80     def __bool__(self) -> "Expr":
---> 81         raise ValueError(
     82             "Since Expr are lazy, the truthiness of an Expr is ambiguous. \
     83             Hint: use '&' or '|' to chain Expr together, not and/or."

ValueError: Since Expr are lazy, the truthiness of an Expr is ambiguous.             Hint: use '&' or '|' to chain Expr together, not and/or.
```

Note that a similar error is thrown when using pl.lit in fill_null for another type (like integer)

**Comment by olivier-lacroix** [comment]
@ritchie46 , I guess the inability to use a pl.Expr in fill_null may be the real issue here, as fill_null is supposed to accept pl.Expr as argument?

**Comment by ritchie46** [comment]
Both fixed in #1756
